src/api/api_auth_shibuqam.nit

```diff
@@ -122,3 +122,51 @@ class ShibCallback
 		return data
 	end
 end
+
+# Information on a user from Shibboleth/UQAM
+class ShUser
+	serialize
+
+	# The *code permanent* (or the uid for non student)
+	var id: String
+
+	# Usually the first name
+	var given_name: String
+
+	# Usually "FamilyName, FirstName"
+	var display_name: String
+
+	# The email @courrier.uqam.ca (or @uqam.ca for non student)
+	var email: String
+
+	# The Gravatar URL (based on `email`)
+	var avatar: String is lazy do
+		var md5 = email.md5
+		return "https://www.gravatar.com/avatar/{md5}?d=retro"
+	end
+end
+
+redef class HttpRequest
+	# Extract the Shibboleth/UQAM information from the header, if any.
+	#
+	# We assume that a reverse proxy does the authentication and fill the request header.
+	# If the server is accessible directly, these headers can be easily forged.
+	# Thus, we assume that the reverse proxy is not by-passable.
+	#
+	# The reverse proxy might choose to force an authentication or not.
+	# If there is no authentication, there is no information in the request header (or with the `(null)` value).
+	# In this case, `null` is returned by this function.
+	fun user: nullable ShUser do
```

Comment:
you do not need this part at all. only the ShUser class.

The protocol is that the oauth server will reply to you with a json object that is unserialized by https://github.com/Morriar/Missions/blob/master/src/api/api_auth_shibuqam.nit#L76

the problem is that the json object will identify itself as a `User`.
A crappy hack could be to force-change the` __class` field of the json object to `ShUser` before unserializing it.